File: ContestOfSharp/ContestOfSharp/Challenges/Challenge0.cs
using System.Reactive.Concurrency;
using ContestOfSharp.North;

namespace ContestOfSharp.Challenges;

public abstract class Challenge0
{
    public static void CountSeconds_Reactively()
    {
        60.SecondsCounter(TaskPoolScheduler.Default)
            .Subscribe(second =>
                Console.WriteLine($"Reactive Timer Ticked: {second}")
            );
    }

    public static async Task CountSeconds_Concurrently()
    {
        await foreach (int second in 60.SecondsCount())
            Console.WriteLine($"Concurrent Timer Ticked: {second}");
    }
}

File: ContestOfSwift/ContestOfSwift/Challenges/Challenge0.swift
//
//  Challenge0.swift
//  ContestOfSwift
//
//  Created by Sergio Daniel on 5/6/24.
//

import Foundation
import Combine

enum Challenge0 {

    static func performReactively() -> AnyCancellable {
        return 60.secondsCounter(on: .main)
            .sink { second in
                print("Reactive Timer Ticked: \(second)")
            }
    }

    @MainActor
    static func performConcurrently() async {
        for await second in 60.secondsCount() {
            print("Concurrent Timer Ticked: \(second)")
        }
    }

}
